lib/src/engine/compressed_texture_formats/ktx_texture.dart

```diff
@@ -0,0 +1,72 @@
+part of stagexl.engine;
+
+// https://registry.khronos.org/KTX/specs/1.0/ktxspec.v1.html
+class KtxFormat {
+  static const ktx_magic = [0xAB, 0x4B, 0x54, 0x58, 0x20, 0x31, 0x31, 0xBB, 0x0D, 0x0A, 0x1A, 0x0A];
+  static const ktx_end_le = 0x04030201;
+  static const ktx_end_be = 0x01020304;
+}
+
+class KtxTexture extends CompressedTexture {
+  late final int _glInternalFormat;
+  late final int _texDataOffset;
+  late final int _imageSize;
+
+  KtxTexture(super.buffer) : super() {
+    _parseHeader();
+  }
+
+  void _parseHeader() {
+    final bytes = ByteArray.fromBuffer(_buffer);
+    final magic = List.generate(12, (_) => bytes.readUnsignedByte());
+
+    for (final entry in magic.asMap().entries) {
+      if (entry.value != KtxFormat.ktx_magic[entry.key]) {
+        print('unrecognized magic, not a ktx header');
+        return;
+      }
+    }
+
+    final endianness = bytes.readUnsignedInt();
+    switch (endianness) {
+      case KtxFormat.ktx_end_le:
+        // We're already reading in little endian
+        break;
+      case KtxFormat.ktx_end_be:
+        // Switch the ByteArray to reading in big endian
+        print('warning: ktx file is big endian, this is untested');
```

Comment:
Maybe add a TODO here.

Reply:
I don't think we'll run into BE files, TexturePacker doesn't create them and neither does Compressonator. I'll mark a TODO to test those in the future though.